transport/graphsync/graphsync.go

```diff
@@ -177,36 +170,28 @@ func (t *Transport) consumeResponses(responseChan <-chan graphsync.ResponseProgr
 }
 
 // Read from the graphsync response and error channels until they are closed
-// or there is an error, then call the channel completed callback
+// or there is an error
 func (t *Transport) executeGsRequest(req *gsReq) {
-	defer req.onComplete()
```

Comment:
I'm not an expert in go-data-transfer, but what replaces the onComplete call here?

Reply:
Now that `graphsync.CancelRequest()` waits until the event queue is drained, we don't need to detect complete manually so we no longer need this callback

Reply:
Actually it is safer to wait for both the events to be drained, and the channel returned from `graphsync.Request()` to be drained. So in the end I think it does make sense to keep the onComplete mechanism 👍 
I've restored it now.